src/os/bluestore/BlueStore.cc

```diff
@@ -1391,6 +1391,10 @@ void BlueStore::BufferSpace::finish_write(Cache* cache, uint64_t seq)
 {
   std::lock_guard<std::recursive_mutex> l(cache->lock);
 
+  if (writing.empty() || writing.rbegin()->seq < seq) {
```

Comment:
(almost irrespective to your commit) It looks like the assumption that entries in 'writing' list are ordered by seq is invalid. E.g. please take a look at _discard() method - it might split Buffer into 2 parts and insert an additional buffer to the end of 'writing' list. Thus the ordering is broken if some entries with higher seq exist. Hence it makes sense to enumerate 'writing' list completely. Otherwise  I suspect (theoretically!) we might leak some entries in this list.